docs/platform-specification.md

```diff
@@ -309,6 +309,59 @@ Sketch uses 924 bytes (2%) of program storage space. Maximum is 32256 bytes.
 Global variables use 9 bytes (0%) of dynamic memory, leaving 2039 bytes for local variables. Maximum is 2048 bytes.
 ```
 
+#### Recipes to compute binary sketch size for more complex systems (since Arduino CLI >=0.21.0)
+
+A platform may provide a tool for the specific purpose to analize the binaries and compute the sketch size and memory
+usage statistics. This is especially useful for boards with non-trivial memory layouts where the classic reg-exp based
+approach is not sufficient.
+
+The command line to run is specified with the recipe **recipe.advanced_size.pattern**.
+
+The expected output from the tool is a JSON object with the following format:
+
+```json
+{
+  "output": "Your sketch use 2200 bytes of program memory out of 8192 (22%)\nThe static RAM used is 200 bytes (of 2048 max)\n",
+  "severity": "info",
+  "sections": [
+    { "name": "text", "size": 2200, "maxsize": 8192 },
```

Comment:
I'm wondering about this `maxsize` key name. It seems most logical to me for the sizer tool output key names to match the machine readable `arduino-cli compile` output, but the key name in the `arduino-cli` output is `max_size`:

```
$ arduino-cli compile -b arduino:avr:uno --format json
{
  "compiler_out": "Sketch uses 444 bytes (1%) of program storage space. Maximum is 32256 bytes.\nGlobal variables use 9 bytes (0%) of dynamic memory, leaving 2039 bytes for local variables. Maximum is 2048 bytes.\n",
  "compiler_err": "",
  "builder_result": {
    "build_path": "C:\\Users\\per\\AppData\\Local\\Temp\\arduino-sketch-5162AF43A5C9EEB136B73635BA92A5B1",
    "executable_sections_size": [
      {
        "name": "text",
        "size": 444,
        "max_size": 32256
      },
      {
        "name": "data",
        "size": 9,
        "max_size": 2048
      }
    ]
  },
  "success": true
}
```

Reply:
Good point I'll change the field name to `max_size`